### PR TITLE
HLRC: ML PUT Calendar

### PR DESCRIPTION
**Pull request by davidkyle**
For #29827

### CODE REVIEW

**Comment by dimitris-athanasiou** [comment]
retest this please

**Review by benwtrent — COMMENTED** [review]
Some minor things and consistencies.

File: client/rest-high-level/src/main/java/org/elasticsearch/client/ml/calendars/Calendar.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+    /**
+     * {@code jobIds} can be a mix of job groups and job Ids
+     * @param id The calendar Id
+     * @param jobIds List of job Ids or job groups
+     * @param description An optional description
+     */
+    public Calendar(String id, List<String> jobIds, @Nullable String description) {
+        this.id = Objects.requireNonNull(id, ID.getPreferredName() + " must not be null");
+        this.jobIds = Objects.requireNonNull(jobIds, JOB_IDS.getPreferredName() + " must not be null");
```
[the remark this answers is not in the document]
Comment (davidkyle) [reply client/rest-high-level/src/main/java/org/elasticsearch/client/ml/calendars/Calendar.java]: Good point

**Review by dimitris-athanasiou — APPROVED** [review]
LGTM

**Pull request merged**